controllers/osbuildenvconfig_controller_test.go

```diff
@@ -322,416 +341,359 @@ var _ = Describe("OSBuildEnvConfig Controller", func() {
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+					// then
+					Expect(err).To(Equal(errFailed))
+					Expect(result).To(Equal(resultRequeue))
+				})
+
+				It("Should return requeue if succeeded to create the Route for the Composer Worker API", func() {
+					// given
+					routeRepository.EXPECT().Create(requestContext, &composerWorkerAPIRoute).Return(nil)
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+					// then
+					Expect(err).To(BeNil())
+					Expect(result).To(Equal(resultRequeue))
+				})
 			})
 
-			Context("Certificate is already created", func() {
+			Context("Route for the Composer Worker API exists ", func() {
 				BeforeEach(func() {
-					certificateRepository.EXPECT().Read(requestContext, composerCertificateName, operatorNamespace).Return(&composerCertificate, nil)
+					routeRepository.EXPECT().Read(requestContext, composerWorkerAPIRouteName, operatorNamespace).Return(&composerWorkerAPIRoute, nil)
 				})
 
-				It("Should return an error if PSQL information is not set", func() {
-					// given
-					psqlError := fmt.Errorf("creating a PSQL service is not yet implemented")
+				It("Should return an error if failed to check if the Route for the Composer Worker API is ready", func() {
+					//given
+					routeRepository.EXPECT().Read(requestContext, composerWorkerAPIRouteName, operatorNamespace).Return(nil, errFailed)
 					// when
 					result, err := reconciler.Reconcile(requestContext, request)
 					// then
-					Expect(err).To(Equal(psqlError))
+					Expect(err).To(Equal(errFailed))
 					Expect(result).To(Equal(resultRequeue))
 
 				})
 
-				Context("PSQL information is set", func() {
+				It("Should return requeue if the Route for the Composer Worker API is not ready", func() {
+					// given
+					composerWorkerAPIRoute.Status = routev1.RouteStatus{
+						Ingress: []routev1.RouteIngress{
+							{
+								Conditions: []routev1.RouteIngressCondition{
+									{
+										Type:   routev1.RouteAdmitted,
+										Status: corev1.ConditionFalse,
+									},
+								},
+							},
+						},
+					}
+					routeRepository.EXPECT().Read(requestContext, composerWorkerAPIRouteName, operatorNamespace).Return(&composerWorkerAPIRoute, nil)
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+					// then
+					Expect(err).To(BeNil())
+					Expect(result).To(Equal(reconcile.Result{Requeue: true, RequeueAfter: time.Second * 10}))
+				})
+
+				Context("Worker API Route is ready", func() {
 					const (
-						composerConfigMapName = "osbuild-composer-config"
+						certificateDuration            = 87600
+						composerComposerAPIServiceName = "osbuild-composer"
+					)
+					var (
+						composerCertificate certmanagerv1.Certificate
 					)
+
 					BeforeEach(func() {
-						var sslMode osbuildv1alpha1.DBSSLMode = "disable"
+						composerWorkerAPIRoute.Status = routev1.RouteStatus{
+							Ingress: []routev1.RouteIngress{
```

Comment:
Maybe the RouteStatus can be part of the on-top beforeEach func and just edit the "status"

Reply:
What's the benefit in that?

Reply:
less duplicated code